my/core/stats.py

```diff
@@ -13,12 +13,12 @@
 
 # TODO maybe could be enough to annotate OUTPUTS or something like that?
 # then stats could just use them as hints?
-def guess_stats(module_name: str) -> Optional[StatsFun]:
+def guess_stats(module_name: str, quick: bool=False) -> Optional[StatsFun]:
```

Comment:
hmm, I feel like `quick` argument belongs to the `auto_stats` below instead? since it's a property of the stats function, not the module itself
then it would be inferred by `inspect.signature` later

Reply:
Oh, I guess it could be -- but that also means the user has to do what happens in `doctor` with the `inspect.signature` (which isn't obvious if they're don't look at the code there)  I was just giving the user more options in case they were calling guess_stats manually -- they could force quick to be true by doing:

```
guess_stats(mod, quick=True)
```

or they could:

```
with quick_stats():
    guess_stats(mod)
```

since in that case in `_stat_iterable` would then look at the global `QUICK_STATS`

Reply:
ah, makese sense. I thought of `guess_stats` is kind of for internal usage tbh? but don't mind either way, just curious what use do you see outside of the doctor

Reply:
yeah would agree its mostly for internal use, but I could see someone wanting to make a custom script which uses only particular modules to test the functionality of

e.g. if I wanted to use doctor on a particular subset of modules programmatically by picking which `my` namespace package

```

In [26]: for mod in list(_discover_path_importables(list(sorted(my.__path__))[3], "my")):
    ...:     mm = importlib.import_module(mod.name)
    ...:     stats_func = guess_stats(mod.name, quick=True)
    ...:     print(stats_func())
    ...: 
{'transactions': {'transactions': {'count': '100+'}}}
{'history': {'history': {'count': '100+'}}, 'to_listen': {'to_listen': {'count': '100+'}}}
{'achievements': {'achievements': {'count': 26, 'last': datetime.datetime(2021, 1, 18, 20, 53)}}, 'forum_posts': {'forum_posts': {'count': '100+', 'last': datetime.datetime(2014, 7, 17, 18, 2, 57, tzinfo=datetime.timezone.utc)}}}
{'locations': {'locations': {'count': '100+', 'last': datetime.datetime(2021, 4, 26, 4, 1, 42, 766000, tzinfo=datetime.timezone.utc)}}}
```

Could always do a bash script and list stuff by then converting to something like:

```
$ fd --glob '*.py' my | tr '/' '.' | rcut -d'.' -f2- | xargs hpi doctor -Sq
✅ OK  : my.location.all                                   
✅     - stats: {'locations': {'count': '100+', 'last': datetime.datetime(2021, 4, 26, 4, 1, 42, 766000, tzinfo=datetime.timezone.utc)}}
✅ OK  : my.mint                                           
✅     - stats: {'_all_balances': {'count': '100+', 'last': datetime.datetime(2020, 10, 20, 12, 5, 52, tzinfo=<git.objects.util.tzoffset object at 0x7f64c7edc340>)}, 'transactions': {'count': '100+'}}
✅ OK  : my.nextalbums                                     
✅     - stats: {'history': {'count': '100+'}, 'to_listen': {'count': '100+'}}
✅ OK  : my.old_forums                                     
✅     - stats: {'forum_posts': {'count': '100+', 'last': datetime.datetime(2014, 7, 17, 18, 2, 57, tzinfo=datetime.timezone.utc)}, 'achievements': {'count': 26, 'last': datetime.datetime(2021, 1, 18, 20, 53)}}
```

But thought might as well make it a bit more user friendly if someone wanted to use it manually for a couple modules from python instead of from the cli